python/tests/test_metadata.py

```diff
@@ -542,6 +542,26 @@ def test_bad_type_error(self):
         ):
             ms.validate_and_encode_row(tskit.TableCollection(1))
 
+    def test_skip_validation(self):
+        ms = tskit.MetadataSchema({"codec": "json"})
+        assert ms._bypass_validation
+        triggered = False
+
+        def tracer(data):
+            nonlocal triggered
+            triggered = True
+
+        ms._validate_row = tracer
+        ms.validate_and_encode_row({})
+        assert not triggered
+
+        ms = tskit.MetadataSchema({"codec": "json", "properties": {"foo": {}}})
+        assert not ms._bypass_validation
+        triggered = False
+        ms._validate_row = tracer
+        ms.validate_and_encode_row({})
+        assert triggered
+
```

Comment:
We should also check that validate is always called on non-json codecs right?

Reply:
Done